### PR TITLE
test: [M3-8393] - Cypress test for Account Maintenance CSV downloads

### PR DESCRIPTION
**Pull request by subsingh-akamai**
## Description 📝
Added cypress test for validating downloaded csv content of account maintenance with Pending and Completed status 

## Changes  🔄
Added below cypress tests under in spec file - cypress/e2e/core/account/account-maintenance.spec.ts
- confirm maintenance details in the tables

## Jira
- M3-8393

## How to test 🧪
`yarn cy:run -s cypress/e2e/core/account/account-maintenance.spec.ts`

### Verification steps
When test executed using this command  `yarn cy:run -s cypress/e2e/core/account/account-maintenance.spec.ts`all tests should pass successfully and you should see new test added as per below screenshot 

image

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by coliu-akamai — COMMENTED** [review]
mainly curious about the new papaparse package this introduces - the last version was released ~1.5 years ago so I'm wondering about its maintainability (although there have been code updates as recent as ~2 months ago). On the other hand, the use cases here look small enough and this package doesn't seem to have other dependencies - @jdamore-linode what are your thoughts?

File: packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts [hunk 1]
```diff
@@ -118,12 +119,135 @@ describe('Maintenance', () => {
...
+          const actualPendingMigrationCsvContent = parsedCsvPendingMigration.map(
+            (entry: any) => ({
+              entity_label: entry['Entity Label'],
+              entity_type: entry['Entity Type'],
+              type: entry['Type'],
+              status: entry['Status'],
+              reason: entry['Reason'],
+            })
+          );
+          console.log('ExpectedCsvContent: ', expectedPendingMigrationContent);
```
[the remark this answers is not in the document]
Comment (subsingh-akamai) [reply packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts]: Resolved in this commit - https://github.com/linode/manager/pull/11168/commits/f31498ae4260e3888eb6f5d7990c142c7d575d04

File: packages/manager/package.json [hunk 2]
```diff
@@ -59,6 +60,7 @@
     "md5": "^2.2.1",
     "notistack": "^3.0.1",
     "qrcode.react": "^0.8.0",
+    "papaparse": "^5.3.0",
```
[the remark this answers is not in the document]
Comment (subsingh-akamai) [reply packages/manager/package.json]: - Removed library - papaparse to parse csv 
- Implemented script to parse the csv file 
 
Commit - https://github.com/linode/manager/pull/11168/commits/d14241e2d3477d6e3f97858a80221ba772061bf8

**Review by coliu-akamai — APPROVED** [review]
thanks @subsingh-akamai! 

✅ confirmed tests pass

File: packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts [hunk 3]
```diff
@@ -118,12 +119,140 @@ describe('Maintenance', () => {
...
+            type: maintenance.type,
+            status: maintenance.status,
+            reason: maintenance.reason,
+          })
+        );
+
+        // Read the downloaded CSV and compare its content to the expected CSV content
+        cy.readFile(`${downloadsFolder}/${fileName}`)
+          .then((csvContent) => {
+            const parsedCsvPendingMigration = parseCsv(csvContent);
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts:158]: I'm noticing an intermittent failure here (and later in the spec when we download the completed maintenance CSV) where `csvContent` is apparently `''`. It seems like there's a brief moment where the CSV file exists but is empty, and an empty string gets passed to the `parseCsv` function which throws an error.

I read a bit into `cy.readFile` [here](https://docs.cypress.io/api/commands/readfile#Automatic-retries), and it looks like adding some assertions on the file contents allows Cypress to wait for the data to actually show up and seems to resolve the issue:

```suggestion
        cy.readFile(`${downloadsFolder}/${fileName}`)
          .should('not.eq', null)
          .should('not.eq', '')
          .then((csvContent) => {
            const parsedCsvPendingMigration = parseCsv(csvContent);
```

Wondering if you ran into this issue at all? My machine is really slow and this issue seems to be pretty time sensitive so I wonder if this occurs at all for developers with newer/faster machines.

**Review by jdamore-linode — APPROVED** [review]
Thanks Subodh, this is looking really nice! I posted a couple suggestions, curious to hear your thoughts!

File: packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts (hunk 3 above)
Comment (coliu-akamai) [reply packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts:158]: I'm on an M3 chip, wasn't running into any issues

**Review by coliu-akamai — APPROVED** [review]
thank you! approving pending Joe's comments reviewed

File: packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts [hunk 4]
```diff
@@ -118,12 +119,140 @@ describe('Maintenance', () => {
...
+        const downloadsFolder = Cypress.config('downloadsFolder');
+
+        // Delete the file before the test
+        cy.exec(`rm -f ${downloadsFolder}/${fileName}`).then((result) => {
+          if (result.code === 0) {
+            cy.log(`Deleted file: ${fileName}`);
+          } else {
+            cy.log(`Failed to delete file: ${fileName}`);
+          }
+        });
```
[the remark this answers is not in the document]
Comment (subsingh-akamai) [reply packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts]: Moved code to delete file using regular expression (pending-maintenance*,completed-maintenance*) and also added failOnNonZeroExit: false which will ensure test will not fail if such file doesn't exist. 

Commit - https://github.com/linode/manager/pull/11168/commits/00ff32cd8b8d67877b8aedaf174dbf39d56a6839

File: packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts [hunk 5]
```diff
@@ -118,12 +119,140 @@ describe('Maintenance', () => {
...
+          .then(() => {
+            // Delete the file after the assertions
+            cy.exec(`rm -f ${downloadsFolder}/${fileName}`).then((result) => {
+              if (result.code === 0) {
+                cy.log(`Deleted file: ${fileName}`);
+              } else {
+                cy.log(`Failed to delete file: ${fileName}`);
+              }
+            });
+          });
```
[the remark this answers is not in the document]
Comment (subsingh-akamai) [reply packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts]: Agreed with your comment, for debugging it make sense not to delete files after execution, made the change for same 
Commit - https://github.com/linode/manager/pull/11168/commits/00ff32cd8b8d67877b8aedaf174dbf39d56a6839

File: packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts (hunk 3 above)
Comment (subsingh-akamai) [reply packages/manager/cypress/e2e/core/account/account-maintenance.spec.ts]: same for me, I didn't came across failure as of now on my machine. Thanks for pointing it out, I did suggested change in this commit https://github.com/linode/manager/pull/11168/commits/764fed390f63f679fd0abf96dcf9f13cb4e5f389

**Review by jdamore-linode — APPROVED** [review]
Thanks @subsingh-akamai, looks great! You're good to disregard those test failures since they're unrelated to your PR's changes

**Pull request merged**